stable/grafana/templates/job.yaml

```diff
@@ -1,50 +1,55 @@
 {{- if .Values.server.setDatasource.enabled -}}
-apiVersion: batch/v1
-kind: Job
-metadata:
-  annotations:
-    "helm.sh/hook": post-install
-    "helm.sh/hook-weight": "-5"
-    "helm.sh/hook-delete-policy": hook-succeeded
-  labels:
-    app: {{ template "grafana.fullname" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{ .Values.server.name }}"
-    heritage: "{{ .Release.Service }}"
-    release: "{{ .Release.Name }}"
-  name: {{ template "grafana.server.fullname" . }}-set-datasource
-spec:
-  activeDeadlineSeconds: {{ .Values.server.setDatasource.activeDeadlineSeconds }}
-  template:
-    metadata:
-      labels:
-        app: {{ template "grafana.fullname" . }}
-        component: "{{ .Values.server.name }}"
-        release: "{{ .Release.Name }}"
-    spec:
-      containers:
-      - name: {{ template "grafana.server.fullname" . }}-set-datasource
-        image: "{{ .Values.server.setDatasource.image }}"
-        env:
-        - name: ADMIN_USER
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "grafana.server.fullname" . }}
-              key: grafana-admin-user
-        - name: ADMIN_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: {{ template "grafana.server.fullname" . }}
-              key: grafana-admin-password
-        args:
-          - "http://$(ADMIN_USER):$(ADMIN_PASSWORD)@{{ template "grafana.fullname" . }}:{{ .Values.server.service.httpPort }}/api/datasources"
-          - "--max-time"
-          - "10"
-          - "-H"
-          - "Content-Type: application/json;charset=UTF-8"
-          - "--data-binary"
-          {{- with .Values.server.setDatasource.datasource }}
-          - "{\"name\":\"{{ .name }}\",\"type\":\"{{ .type }}\",\"url\":\"{{ .url }}\",\"database\":\"{{ .database }}\",\"jsonData\":{ {{ .jsonData }} },\"access\":\"{{ .access }}\",\"isDefault\":{{ .isDefault }}}"
-          {{- end }}
-      restartPolicy: {{ .Values.server.setDatasource.restartPolicy }}
+apiVersion: v1
+kind: List
```

Comment:
There's no need to create a `List`. You can just separate the resources  with `---`.